server/rest/src/test/java/org/infinispan/rest/resources/metedata/EntryVersionMetadataCollectionTest.java

```diff
@@ -0,0 +1,128 @@
+package org.infinispan.rest.resources.metedata;
+
+import static org.infinispan.commons.dataconversion.MediaType.TEXT_PLAIN_TYPE;
+import static org.infinispan.rest.assertion.ResponseAssertion.assertThat;
+import static org.infinispan.util.concurrent.CompletionStages.join;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.infinispan.AdvancedCache;
+import org.infinispan.Cache;
+import org.infinispan.client.rest.RestCacheClient;
+import org.infinispan.client.rest.RestClient;
+import org.infinispan.client.rest.RestResponse;
+import org.infinispan.client.rest.configuration.RestClientConfigurationBuilder;
+import org.infinispan.commons.dataconversion.internal.Json;
+import org.infinispan.commons.test.TestResourceTracker;
+import org.infinispan.commons.test.annotation.TestForIssue;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.container.versioning.NumericVersion;
+import org.infinispan.container.versioning.SimpleClusteredVersion;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.metadata.EmbeddedMetadata;
+import org.infinispan.metadata.Metadata;
+import org.infinispan.rest.helper.RestServerHelper;
+import org.infinispan.test.SingleCacheManagerTest;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.testng.annotations.Test;
+
+@Test(groups = "unit", testName = "rest.search.resources.metedata.EntryVersionMetadataCollectionTest")
+@TestForIssue(jiraKey = "ISPN-14788")
+public class EntryVersionMetadataCollectionTest extends SingleCacheManagerTest {
```

Comment:
you don't need a new class. `org.infinispan.rest.resources.CacheResourceV2Test` tests all the methods related to the cache endpoint and the test can live there.

Reply:
My idea was try to start to offload some code from `CacheResourceV2Test` class that has 1788 locs, but it is ok to add a method there.